quint/src/names/base.ts

```diff
@@ -68,6 +68,19 @@ export type DefinitionsByModule = Map<string, DefinitionsByName>
  */
 export type LookupTable = Map<bigint, LookupDefinition>
 
+/**
+ * A lazy mapping from module names to the definitions that are available but not used in that module.
```

Comment:
Because it is a function, and therefore will only be evaluated if called. The value here is a set difference between all names in a module and the referenced names, so I imagine is not a super cheap operation - and we only need to use this mapping for one specific module (the main module, which we don't know what it is at this point) and in one specific cli command (test).

The strict version would be to define a `Map` instead of a function, at least in my understanding.